packages/api-client/src/libs/send-request/set-request-cookies.ts

```diff
@@ -0,0 +1,166 @@
+ *
+ * None specifies that cookies are sent on both originating and cross-site requests.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#controlling_third-party_cookies_with_samesite
+ */
+const defaultSameSite = 'None'
+
+/**
+ * Set all cookie params and workspace level cookies that are applicable
+ */
+export function setRequestCookies({
+  example,
+  env,
+  globalCookies,
+  serverUrl,
+  proxyUrl,
+}: {
+  example: RequestExample
+  env: object
+  globalCookies: Cookie[]
+  serverUrl: string
+  proxyUrl?: string
+}): {
+  cookieParams: Cookie[]
+} {
+  const cookieParams: Cookie[] = []
+
+  const isUsingTheProxy = shouldUseProxy(proxyUrl, serverUrl)
+
+  // Try to add the target domain with a wildcard dot
+  const defaultDomain = determineCookieDomain(
+    isUsingTheProxy ? (proxyUrl as string) : (serverUrl ?? 'http://localhost'),
+  )
+
+  // Add global cookies that match the current domain
+  globalCookies.forEach((c) => {
+    const { name, value, domain: configuredHostname, ...params } = c
+
+    if (!matchesDomain(serverUrl, configuredHostname)) {
+      return
+    }
+
+    if (!name) {
+      return
+    }
+
+    cookieParams.push({
+      uid: name,
+      name,
+      value,
+      domain: configuredHostname,
+      path: params.path,
+    })
+  })
+
+  // Add local cookies
+  example.parameters.cookies.forEach((c) => {
+    if (!c.enabled) {
+      return
+    }
+
+    if (!c.key) {
+      return
+    }
+
+    cookieParams.push({
+      uid: c.key,
+      name: c.key,
+      value: replaceTemplateVariables(c.value, env),
+      domain: defaultDomain,
+      path: defaultPath,
+    })
+  })
+
+  return {
+    cookieParams,
+  }
+}
+
+/**
+ * If the Set-Cookie header does not specify a Domain attribute, the cookies are available on the server that sets it
+ * but not on its subdomains. Therefore, specifying Domain is less restrictive than omitting it.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies
+ */
+const determineCookieDomain = (url: string) => {
+  const hostname = new URL(url.startsWith('http') ? url : `http://${url}`)
+    .hostname
+
+  // If it’s an IP, just return it
+  if (hostname.match(/^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/)) {
+    return hostname
+  }
+
+  // If it’s IPv6, just return it
+  if (hostname.match(/^[a-fA-F0-9:]+$/)) {
+    return hostname
+  }
+
+  // If it’s a hostname, return it with a dot
+  return hostname.startsWith('.') ? hostname : `.${hostname}`
+}
+
+/**
+ * Matches, when:
+ * - Isn’t scoped to a domain, or
+ * - matches the current host, or
+ * - or ends with the current host, or
+ * - matches the current host with a wildcard.
+ */
+export const matchesDomain = (
+  givenUrl?: string,
+  configuredHostname?: string,
+): boolean => {
+  if (!givenUrl || !configuredHostname) return true
+
+  try {
+    const givenHostname = new URL(
+      givenUrl.startsWith('http') ? givenUrl : `http://${givenUrl}`,
+    ).hostname
+
+    return (
```

Comment:
😅

Reply:
we can break this one up into vars, might be easier to understand

Reply:
or maybe just add a comment

Reply:
true!